lib/trino-orc/src/main/java/io/trino/orc/metadata/statistics/TimeStatisticsBuilder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.orc.metadata.statistics;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.type.Type;
+
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+
+public class TimeStatisticsBuilder
+        extends IntegerStatisticsBuilder
```

Comment:
We usually avoid using inheritance to share code. This should implement LongValueStatisticsBuilder along similar lines as TimestampStatisticsBuilder